### ISSUE TITLE
Segmentation fault in Magnum::GL::AbstractTexture::bind(int) in ImGui application after loading a font

### ISSUE THREAD

**Issue by Xeverous** [mosra/magnum#476]
I have modified ImGui example application, adding following lines to platform constructor:

```cpp
	_imgui = Magnum::ImGuiIntegration::Context([...]; // already present line

	// added code:
	ImGuiIO& io = ImGui::GetIO();
	_fontin_small_caps = io.Fonts->AddFontFromFileTTF("assets/fonts/Fontin-SmallCaps.otf", 24);
	_fontin_regular    = io.Fonts->AddFontFromFileTTF("assets/fonts/Fontin-Regular.otf", 24);
	IM_ASSERT(_fontin_small_caps);
	IM_ASSERT(_fontin_regular_caps);

	io.Fonts->Build(); // X
	IM_ASSERT(_fontin_small_caps->IsLoaded()); // X
	IM_ASSERT(_fontin_regular_caps->IsLoaded()); // X
	io.FontDefault = _fontin_regular; // X

	// renderer blend and minimal loop period further...
```

All of the added code is pretty much copy-pasted from https://github.com/ocornut/imgui/blob/master/docs/FONTS.md

Running on Windows (haven't tried elsewhere), the program crashes immediately upon startup:

```
Thread 1 received signal SIGSEGV, Segmentation fault.
(gdb) bt
#0  0x000000006334dc03 in Magnum::GL::AbstractTexture::bind(int) ()
#1  0x000000006df920de in Magnum::Shaders::Flat<2u>::bindTexture(Magnum::GL::Texture<2u>&) ()
#2  0x00000000628826f4 in Magnum::ImGuiIntegration::Context::drawFrame() ()
#3  0x0000000000403bbf in Magnum::Examples::Application::drawEvent() ()
#4  0x0000000000404a37 in Magnum::Platform::Sdl2Application::mainLoopIteration() [clone .part.0] ()
#5  0x00000000004062cf in Magnum::Platform::Sdl2Application::exec() ()
#6  0x000000000040f016 in main ()
```

If I comment out all 4 lines marked with X comment, then:

- the executable does not crash
- the UI uses built-in ImGui font
- I can go to `Demo Window->Tools->Style Editor->Fonts` and select newly loaded font as default and everything works

So in summary, the crash happens only if immediately after loading a font I try to build it and set it as the default font.

**Comment by Xeverous** [comment]
Wait please, I just discovered https://doc.magnum.graphics/magnum/classMagnum_1_1ImGuiIntegration_1_1Context.html#ImGuiIntegration-Context-fonts and looks like it's completely my fault!

**Comment by Xeverous** [comment]
Yeah, that was my fault. I just wan't aware there are some preconditions mentioned in context class documentation.

**Comment by mosra** [comment]
Huh, what the heck, even if you didn't read any of the ImGuiIntegration docs, it *should not* crash like this. I have to check what's going on, there has to be a more graceful way to handle this than crashing for no reason in this particular function.

By the way, for general questions and such you might want to join https://gitter.im/mosra/magnum (simply log in with your GitHub account) ;)

**Comment by mosra** [comment]
As of mosra/magnum-integration@1117d6f9b843aafe1fbd95bbadbc36e1180a488a the ImGui integration uses raw OpenGL IDs inside `ImTextureID` instead of a pointer to a live `GL::Texture2D` instance, which should prevent nasty crashes like this.
